File: C#/1087 Dama.cs
using System;
class HelloWorld {
  static void Main() {
    /*NÃO ESTA CORRETO*/
   	string[] s = Console.ReadLine().Split(); bool verificar;
   	int x1 = int.Parse(s[0]); 
   	int y1 = int.Parse(s[1]); 
   	int x2 = int.Parse(s[2]);
   	int y2 = int.Parse(s[3]);
   	while(x1!=0 && x2!=0 && y1!=0 && y2!=0){
   	    if(x1 == x2 && y1 == y2) Console.WriteLine(0);
   	    else{
   	        verificar = false;
   	        if(x1 == x2 || y1==y2) verificar = true;
    		if(Math.Abs(x1-x2)==Math.Abs(y1-y2)) verificar = true;
    		if(verificar) Console.WriteLine(1);
    		else Console.WriteLine(2);
    		s = Console.ReadLine().Split();
    		x1 = int.Parse(s[0]); 
       	    y1 = int.Parse(s[1]); 
       	    x2 = int.Parse(s[2]);
       	    y2 = int.Parse(s[3]);
   	    }
   	}
  }
}

File: C++/1087 Dama.cpp
#include <iostream>
#include <cmath>

using namespace std;
int main() {
    int x1, y1, x2, y2;
    cin >> x1 >> y1 >> x2 >> y2;
    while(x1 !=0 and x2 != 0 and y1 !=0 and y2!=0){
	
    	if(x1 == x2 and y1 == y2) cout << 0 << endl;
    	else{
    		bool verificar = false;
    		if(x1 == x2 or y1==y2) verificar = true;
    		if(abs(x1-x2)==abs(y1-y2)) verificar = true;
    		if(verificar) cout << 1 << endl;
    		else cout << 2 << endl;
		}
    	
		cin >> x1 >> y1 >> x2 >> y2;
   	}
    return 0;
}
